src/mo_frontend/typing.ml

```diff
@@ -1528,6 +1459,12 @@ and infer_exp'' env exp : T.typ =
       | Some typ -> typ
       | None -> {it = TupT []; at = no_region; note = T.Pre}
     in
+    begin match exp1.it with
+    | AsyncE (Some par, _, _, _) when not env.pre && T.is_shared_sort shared_pat.it ->
+      local_error env par.at "M0213"
+        "parenthetical notes aren't allowed on public methods"
```

Comment:
```suggestion
        "parenthetical notes aren't allowed on shared functions"
```

Hmm, are they allowed on local functions?
```
func f() : async Nat = (with cycles =...) async { .. }
```
I guess that might actually work but just flagging in case you meant to rule it out.

Reply:
This is actually intended. A minor feature if you will. It shields the send from a caller-side parenthetical, and can impose modalities locally instead. Normally people will use the `func() : async () { ... }` syntax which just accepts the modalities of the sender side.